### DISCUSSION TITLE
Custom plugin, ButtonsColumn and namespace errors

### DISCUSSION THREAD

**Discussion by cpmills1975** [netbox-community/netbox#7390] [Q&A] [answered]
I suspect I'm going to be told I'm dabbling with private classes and should steer well clear, but here goes...

I'm working on a custom plugin for NetBox 3.0.3. The custom plugin presents a list of objects and I'd like to present a familiar UI/UX as the other built in lists.

I've defined a table class inheriting from BaseTable and I'm able to render it using a subclass of generic.ObjectListView. I tried to add a ButtonsColumn(MyModel) so I could have the standard set of changelog, edit and delete buttons and then I hit problems with 'plugin_name is not a registered namespace' errors. I tracked this down to the template code that renders the changelog, edit and delete buttons and, with plenty of googling, managed to work out that my plugin's namespace was 'plugins:plugin_name' not just 'plugin_name'. Consequently the template was unable to render my links and fell over.

I've been able to workaround this by subclassing the ButtonColumn class and adding 'plugins:' before the  {app_label}:{model_name} in the links, but this feels clunky. Is this my only option? Can I/should I somehow be changing the app_label?

I'd very much like to keep my UX consistent with the native tables and therefore just create suitably named views that can be called by the 'built in' templates if that is somehow possible. Is there a bug or a FR I could raise here to make this available? Is my only option here to re-invent the wheel and craft my own template code to display these buttons?

Many thanks.

**Comment by DanSheps** [comment, marked as answer]
Check out my netbox-plugin-extensions to see how I handled that.

I think ultimately the goal is to make the helper function plugin safe, but that likely won't be until 3.1

**Reply by hagbarddenstore** [reply]
I highly suggest using my fork of Dans plugin, because as of now, Dans plugin is a "bit" broken... https://github.com/DanSheps/netbox-plugin-extensions/pull/4

**Reply by DanSheps** [reply]
What specifically is broken with it?

If you are doing something that doesn't work with my plugin, you are probably pretty far outside of what is "allowable".  My plugin is already outside of what we prefer people to do but is meant as a stopgap until some of the helper functions can be extended within the NetBox core.

**Reply by cpmills1975** [reply]
@DanSheps thanks, I see you are basically creating your own versions of these templates. It would be awesome if plugins could leverage definitive versions of these templates from the code but in the mean time, I'll replicate what I need. Many thanks.

**Reply by DanSheps** [reply]
Feel free to let me know if you need something specific
